### PR TITLE
Add restrictions for orioledb technical preview

### PR DESCRIPTION
**Pull request by joshenlim**
Dependent on API, but not blocked by
https://github.com/supabase/infrastructure/pull/20480

### CODE REVIEW

File: apps/studio/hooks/misc/useSelectedOrganization.ts [hunk 1]
```diff
@@ -12,6 +36,7 @@ export function useSelectedOrganization({ enabled = true } = {}) {
   const { data } = useOrganizationsQuery({ enabled: isLoggedIn && enabled })
 
   const selectedProject = useProjectByRef(ref)
+  console.log({ selectedProject })
```
Comment (alaister) [inline apps/studio/hooks/misc/useSelectedOrganization.ts:39]:
```suggestion
```

File: apps/studio/hooks/misc/useSelectedOrganization.ts [hunk 2]
```diff
@@ -4,6 +4,30 @@ import { useMemo } from 'react'
...
+  const orgSlug = slug ?? selectedProject?.organization_slug ?? localStorageSlug
+  const { data } = useOrganizationQuery(
+    { slug: orgSlug as string },
+    { enabled: enabled && isLoggedIn && typeof orgSlug === 'string' }
+  )
+
+  return useMemo(() => {
+    return data
+  }, [data, slug, selectedProject, localStorageSlug])
+}
```
Comment (alaister) [inline apps/studio/hooks/misc/useSelectedOrganization.ts:30]: Is this used in this PR?

File: apps/studio/components/interfaces/Organization/BillingSettings/Subscription/PlanUpdateSidePanel.tsx [hunk 3]
```diff
@@ -60,6 +60,10 @@ const PlanUpdateSidePanel = () => {
     (it) => it.organization_id === selectedOrganization?.id
   )
 
+  const { data } = useOrganizationQuery({ slug })
+  console.log({ selectedOrganization, data })
```
Comment (alaister) [inline apps/studio/components/interfaces/Organization/BillingSettings/Subscription/PlanUpdateSidePanel.tsx:64]:
```suggestion
```

File: apps/studio/hooks/misc/useSelectedOrganization.ts (hunk 2 above)
Comment (joshenlim) [reply apps/studio/hooks/misc/useSelectedOrganization.ts:30]: not yet - but i think i'll look into seeing whether we can use this in a subsequent PR, will keep this PR strictly just for oriole DB changes to make things easier to review and test

**Review by staaldraad — APPROVED** [review]

**Review by alaister — APPROVED** [review]
LGTM!

- Tested locally as described
- Tested locally with a regular project to ensure everything still works

**Comment by pcnc** [comment]
Seems that the oriole warning also appears when oriole is not available (e.g for paid orgs)

image
image

**Comment by pcnc** [comment]
image
image

File: apps/studio/components/interfaces/Organization/BillingSettings/Subscription/PlanUpdateSidePanel.tsx [hunk 4]
```diff
@@ -251,17 +253,24 @@ const PlanUpdateSidePanel = () => {
...
                             side: 'bottom',
+                            className: hasOrioleProjects ? 'w-80 text-center' : '',
                             text:
                               subscription?.plan?.id === 'enterprise'
                                 ? 'Reach out to us via support to update your plan from Enterprise'
-                                : !canUpdateSubscription
-                                  ? 'You do not have permission to change the subscription plan'
-                                  : undefined,
+                                : hasOrioleProjects
+                                  ? 'Your organization has projects that are on OrioleDB, which is only for the free plan. Remove all OrioleDB projects before changing your plan'
```
Comment (w3b6x9) [inline apps/studio/components/interfaces/Organization/BillingSettings/Subscription/PlanUpdateSidePanel.tsx:270]:
```suggestion
                                  ? 'Your organization has projects that are using the OrioleDB extension which is only available on the Free plan. Remove all OrioleDB projects before changing your plan'
```

**Review by alaister — APPROVED** [review]
LGTM!

- Tested on preview:
  - Creating an oriole project
  - Creating a regular project in a free org
  - Creating a regular project in a paid org
  - Realtime badge
  - cmd+i working

**Pull request merged**